### ISSUE TITLE
Factor out Daemon / Gateway endpoints handlers 

### ISSUE THREAD

**Issue by Gozala** [ipfs/js-ipfs#1855]
- **Version**: v0.33.1
- **Platform**: Browser
- **Subsystem**: Workers / WebExtensions

#### Type:
Feature
#### Severity:
Medium
## TL;DR

IPFS SW should use local IPFS node on the system and transparently fallback to in-browser IPFS node when local node isn't available.

## Overview

I am working towards makeing in-browser IPFS exposed through SW compatible with a REST API exposed by local IPFS daemon / gateway. For more details see https://github.com/ipfs/in-web-browsers/issues/137

There is already [code in place](https://github.com/ipfs/js-ipfs/tree/master/src/http) that binds API endpoints to IPFS API calls, but it's not available in browser dist and there is an assumption about running in nodejs using hapi.

## Proposal

I would like to propose to factor out API endpoint binding code  into separate lib or maybe just bundle and in a way that would remove node / hapi assumptions, such that would allow us to share that codebase across SW and nodejs contexts.

**Comment by Gozala** [comment]
There is obvious overlap with #1820, it maybe that just expanding it's context would make more sense than having this as separate issue. However it was fairly easy to address issue pointed out there using https://github.com/ipfs/js-ipfs-http-response but dealing with Daemon REST API continues to be quite painful due to larger API surface.

**Comment by alanshaw** [comment]
As a data point, `require('src/http/routes')` gives you an array of route handlers that look something like this for example:

```js
{
  method: '*',
  path: '/api/v0/dns',
  handler: async (request, h) { /* ... */ }
}
```

There should be no dependencies on `Hapi` in this tree but it does use `joi` for validation and `boom` for creating http friendly error messages.

So, in that sense if we just extracted `src/http` to a seperate module as-is, you _could_ share it without further work if you built an adapter to map whatever request system you're using to them.

**Comment by SgtPooki** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).

Please feel to reopen with any comments by 2023-06-02. We will do a final pass on reopened issues afterwards (see https://github.com/ipfs/js-ipfs/issues/4336).

This issue will most likely be handled as we improve on Helia, but marking as "wontfix" for now for the sake of analyzing any of these js-ipfs deprecation issues.
